libqtile/backend/wayland/cffi/build.py

```diff
@@ -166,30 +168,34 @@ def get_include_path(lib: str) -> str:
     QW_PROTO_OUT_PATH,
 ]
 LIBRARIES = ["wlroots-0.19", "wayland-server", "input", "cairo"]
-LIBWAYC = QW_PATH / ("..") / "libwayc.so"
+SOURCE_FILES = glob.glob(f"{QW_PATH}/*.c")
+OBJECTS = [Path(src).parent / "build" / Path(src).with_suffix(".o").name for src in SOURCE_FILES]
 
 ffi = FFI()
 ffi.set_source(
     "libqtile.backend.wayland._ffi",
     SOURCE,
     define_macros=[("WLR_USE_UNSTABLE", None)],
     include_dirs=INCLUDE_DIRS,
-    extra_link_args=[LIBWAYC],
+    extra_objects=OBJECTS,
+    libraries=LIBRARIES,
 )
 
-# Create a Makefile include to pass parameters for building wayc as library
-with open(QW_PATH / ".." / "cffi" / "libconfig.mk", "w") as f:
-    f.write("# Auto-generated by generated by cffi/build.py\n\n")
-    f.write(f"QW_PATH := {QW_PATH}\n")
-    f.write(f"QW_PROTO_OUT := {QW_PROTO_OUT_PATH}\n")
-    f.write(f"CC := {os.getenv(sysconfig.get_config_var('CC'), 'gcc')}\n")
-    f.write(f"INCLUDES := {' '.join([f'-I{path}' for path in INCLUDE_DIRS])}\n")
-    f.write(f"LDLIBS := {' '.join([f'-l{lib}' for lib in LIBRARIES])}\n")
-    f.write(f"TARGET := {LIBWAYC}\n")
 
+def build_objects() -> None:
+    # We have to use relative paths here for output_dir to work as expected
+    os.chdir(QW_PATH)
```

Comment:
This chdir is currently causing a build issue, as we no longer are able to write `libqtile/_build_config.py`, since we moved